api/envoy/extensions/filters/http/ratelimit/v3/rate_limit.proto

```diff
@@ -117,6 +361,28 @@ message RateLimitPerRoute {
     IGNORE = 2;
   }
 
+  // The override option determines how the filter handles the cases where there is an override at multiple levels (virtual host, route, cluster weight).
+  enum OverrideOptions {
+    // Client-defined default, typically OVERRIDE_POLICY. If VhRateLimitsOptions is set, that will be used instead.
+    DEFAULT = 0;
+
+    // Only use the specific rate limit policy override when the route doesn't have a rate limit policy.
```

Comment:
I don't think these comments are quite clear enough.  When you say "the specific" policy, it's not clear whether you're talking about this one or the more specific one at a lower level.  Also, they should not mention "route" at all, since the more specific config may not be in the route (e.g., there may be override configs only at the virtual host and cluster weight levels).

I suggest phrasing this like this:

"""
If there is an override config at a more specific level, use that instead of this one.
"""

Similarly for `INCLUDE_POLICY`:

"""
If there is an override config at a more specific level, use data from both.
"""

And for `IGNORE_POLICY`:

"""
If there is an override config at a more specific level, ignore it and use only this one.
"""

Reply:
I phrase the comment based on reading the implementation code (line#54 -- line#68, also copy&paster below):
`case OverrideOptions::Include:
    populateRateLimitDescriptors(route_entry->virtualHost().rateLimitPolicy(), descriptors,
                                 headers);
    break;
  case OverrideOptions::Override:
    if (route_entry->rateLimitPolicy().empty()) {
      populateRateLimitDescriptors(route_entry->virtualHost().rateLimitPolicy(), descriptors,
                                   headers);
    }
    break;
`
Do you envision that we will update the implementation code in the future to reflect more specific level override?

Reply:
Yes, the whole point of having a new enum here is to make the behavior more generic, so that it's not tied specifically to the virtual host or route levels.  The current configuration *is* tied to those two specific levels, because the configuration exists directly in the `VirtualHost` and `RouteAction` protos rather than being here in the filter config, and there is no corresponding field in the `ClusterWeight` proto.  But when we move the configuration here to the filter config, it will also support the `ClusterWeight` level.

But you raise a good point that this additional flexibility doesn't exist yet.  So I propose that we remove the implementation from this PR and instead implement the new override options later, at the same time that we implement moving the rate limit config itself here into the filter config.  For now, you can add a `[#not-implemented-hide:]` comment on this new `OverrideOptions` enum, and remove the deprecated tag from `VhRateLimitOptions`.

Reply:
Yea. Good idea! It make sense now with future filter config change. Thanks for details!

@mattklein123 , Here is the discussion about if we want to do the implementation change in this PR:
Partial implementation and the deprecation was done in previous commit, but as we looked into this closer.  eve though it/deprecation change works fine from function perspective, we think it is clearer to do the implementation/deprecation in the future PR along with tasks described here #18118.

So there is no implementation change in this PR.

Reply:
OK sounds good. Thanks for doing the implementation work. Happy to help review the follow up.